Improve `hvcat` signature in its docstring

jishnub opened the pull request:
Replace the name of the first argument in the docstring from `rows` to `blocks_per_row`, which is closer to what the argument stands for.

jishnub committed:
improve hvcat signature in docstring

dkarrasch commented on base/abstractarray.jl:
```diff
@@ -2015,9 +2016,6 @@ julia> hvcat((2,2,2), a,b,c,d,e,f)
  3  4
  5  6
```
```suggestion
 5  6
 
julia> hvcat((2,2,2), a,b,c,d,e,f) == hvcat(2, a,b,c,d,e,f)
true
```
? Not sure, because the doctests are already long, but this could demonstrate the single-integer usage.

jishnub committed:
Include single integer example in doctests

Co-authored-by: Daniel Karrasch <daniel.karrasch@posteo.de>

jishnub commented on base/abstractarray.jl:
```diff
@@ -1982,11 +1982,12 @@ function hvcat(nbc::Integer, as...)
 
 """
-    hvcat(rows::Tuple{Vararg{Int}}, values...)
+    hvcat(blocks_per_row::Union{Tuple{Vararg{Int}}, Integer}, values...)
```
should this be `Tuple{Vararg{Integer}}` instead?

dkarrasch replied:
The signatures below indicate it must be `Int`. Also, this fails

```julia
hvcat((Int8(2),Int8(2),Int8(2)), a,b,c,d,e,f)
```

jishnub replied:
Ah perhaps the single integer argument should also be an `Int` in that case, as this fails as well:
```julia
julia> hvcat(Int8(2), 1:6...)
ERROR: MethodError: no method matching hvcat(::Tuple{Int8, Int8, Int8}, ::Int64, ::Int64, ::Int64, ::Int64, ::Int64, ::Int64)
Closest candidates are:
  hvcat(::Integer, ::Any...) at abstractarray.jl:1942
  hvcat(::Tuple{Vararg{Int64}}, ::T...) where T<:Number at abstractarray.jl:2038
  hvcat(::Tuple{Vararg{Int64}}, ::Number...) at abstractarray.jl:2075
  ...
Stacktrace:
 [1] hvcat(::Int8, ::Int64, ::Vararg{Int64})
   @ Base ./abstractarray.jl:1948
 [2] top-level scope
   @ REPL[2]:1
```

jishnub committed:
Change `Integer` to `Int`

The single integer case only works with an `Int`, so this fixes the docstring accordingly.

dkarrasch commented:
If that fails anyway, we should perhaps restrict the type in line 1975 to the type that is working, i.e, `Int`.

jishnub committed:
Update single-integer method signature

Change `Integer` to `Int`, as other `Integer` types don't seem to work.

giordano commented on base/abstractarray.jl:
```diff
@@ -2014,10 +2015,10 @@ julia> hvcat((2,2,2), a,b,c,d,e,f)
  1  2
  3  4
  5  6
+ 
```
```suggestion

```

giordano committed:
Update base/abstractarray.jl
